packages/elements/src/photon-multirx-form/components/AddPrescriptionCard.tsx

```diff
@@ -81,27 +101,30 @@ export const AddPrescriptionCard = (props: {
     props.actions.validate(keys);
     const errorsPresent = props.actions.hasErrors(keys);
     if (!errorsPresent) {
+      const draft = {
+        id: String(Math.random()),
+        effectiveDate: props.store.effectiveDate.value,
+        treatment: props.store.treatment.value,
+        dispenseAsWritten: props.store.dispenseAsWritten.value,
+        dispenseQuantity: props.store.dispenseQuantity.value,
+        dispenseUnit: props.store.dispenseUnit.value,
+        daysSupply: props.store.daysSupply.value,
+        refillsInput: props.store.refillsInput.value,
+        instructions: props.store.instructions.value,
+        notes: props.store.notes.value,
+        fillsAllowed: props.store.refillsInput.value + 1,
+        addToTemplates: props.store.addToTemplates?.value ?? false,
+        templateName: props.store.templateName?.value ?? '',
+        catalogId: props.store.catalogId.value ?? undefined
+      };
+
       props.actions.updateFormValue({
         key: 'draftPrescriptions',
-        value: [
-          ...(props.store.draftPrescriptions?.value || []),
-          {
-            id: String(Math.random()),
-            effectiveDate: props.store.effectiveDate.value,
-            treatment: props.store.treatment.value,
-            dispenseAsWritten: props.store.dispenseAsWritten.value,
-            dispenseQuantity: props.store.dispenseQuantity.value,
-            dispenseUnit: props.store.dispenseUnit.value,
-            daysSupply: props.store.daysSupply.value,
-            refillsInput: props.store.refillsInput.value,
-            instructions: props.store.instructions.value,
-            notes: props.store.notes.value,
-            fillsAllowed: props.store.refillsInput.value + 1,
-            addToTemplates: props.store.addToTemplates?.value ?? false,
-            catalogId: props.store.catalogId.value ?? undefined
-          }
-        ]
+        value: [...(props.store.draftPrescriptions?.value || []), draft]
       });
+
+      const addToTemplate = props.store.addToTemplates?.value ?? false;
```

Comment:
these should be `createMemo` if you're deriving from a reactive variable

Reply:
I don't really know how solid works, but I'd think that I dont because its just part of the `handleAddPrescription`, not part of the render logic. I just need a handle on it before the form is cleared. 

Am I wrong?